samples/vboxwrapper/vbox_vboxmanage.cpp

```diff
@@ -58,821 +58,816 @@ using std::string;
+                    vbm_popen(command, output, "remove virtual ISO 9660 disk", false);
+                    if (enable_cache_disk) {
+                        command  = "closemedium disk \"" + virtual_machine_slot_directory + "/" + cache_disk_filename + "\" ";
+                        vbm_popen(command, output, "remove virtual cache disk", false);
+                    }
                 }
             }
         }
+        return 0;
     }
-    return 0;
-}
 
-int VBOX_VM::poll(bool log_state) {
-    int retval = ERR_EXEC;
-    APP_INIT_DATA aid;
-    string command;
-    string output;
-    string::iterator iter;
-    string vmstate;
-    static string vmstate_old = "poweroff";
-    size_t vmstate_start;
-    size_t vmstate_end;
+    int VBOX_VM::poll(bool log_state) {
+        int retval = ERR_EXEC;
+        APP_INIT_DATA aid;
+        string command;
+        string output;
+        string::iterator iter;
+        string vmstate;
+        static string vmstate_old = "PoweredOff";
 
-    boinc_get_init_data_p(&aid);
+        boinc_get_init_data_p(&aid);
 
-    //
-    // Is our environment still sane?
-    //
+        //
+        // Is our environment still sane?
+        //
 #ifdef _WIN32
-    if (aid.using_sandbox && vboxsvc_pid_handle && !process_exists(vboxsvc_pid_handle)) {
-        vboxlog_msg("Status Report: vboxsvc.exe is no longer running.");
-    }
-    if (started_successfully && vm_pid_handle && !process_exists(vm_pid_handle)) {
-        vboxlog_msg("Status Report: virtualbox.exe/vboxheadless.exe is no longer running.");
-    }
+        if (aid.using_sandbox && vboxsvc_pid_handle && !process_exists(vboxsvc_pid_handle)) {
+            vboxlog_msg("Status Report: vboxsvc.exe is no longer running.");
+        }
+        if (started_successfully && vm_pid_handle && !process_exists(vm_pid_handle)) {
+            vboxlog_msg("Status Report: virtualbox.exe/vboxheadless.exe is no longer running.");
+        }
 #else
-    if (started_successfully && vm_pid && !process_exists(vm_pid)) {
-        vboxlog_msg("Status Report: virtualbox/vboxheadless is no longer running.");
-    }
-#endif
-
-    //
-    // What state is the VM in?
-    //
-
-    command  = "showvminfo \"" + vm_name + "\" ";
-    command += "--machinereadable ";
+        if (started_successfully && vm_pid && !process_exists(vm_pid)) {
+            vboxlog_msg("Status Report: virtualbox/vboxheadless is no longer running.");
+        }
+#endif
 
-    if (vbm_popen(command, output, "VM state", false, false, 45, false) == 0) {
-        vmstate_start = output.find("VMState=\"");
-        if (vmstate_start != string::npos) {
-            vmstate_start += 9;
-            vmstate_end = output.find("\"", vmstate_start);
-            vmstate = output.substr(vmstate_start, vmstate_end - vmstate_start);
+        //
+        // What state is the VM in?
+        //
+        vmstate =read_vm_log();
```

Comment:
This reads the status directly from the VirtualBox log. The subsequent string comparisons are using capitalized names now and where using lower case names before. It would be better to lower case the string extracted from the log and compare with lowercase names to compensate for past or future changes in VirtualBox.

Reply:
The change is because VirtualBox writes the states in VBox.log with this capitalisation. Previously, lowercase was used because VBox uses only lowercase letters for the showvminfo command output (VMstate field). 

The procedure followed by VirtualBox to translate the MachineState:

https://www.virtualbox.org/browser/vbox/trunk/src/VBox/Frontends/VBoxManage/VBoxManageInfo.cpp#L135